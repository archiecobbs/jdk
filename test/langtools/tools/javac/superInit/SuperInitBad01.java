/*
 * @test /nodynamiccopyright/
 * @bug 8193760
 * @summary Test circular invocation of this() in constructors
 *
 * @compile/fail/ref=SuperInitBad01.out -XDrawDiagnostics SuperInitBad01.java
 */
class SuperInitBad01 {

    SuperInitBad01(double x) {
        if (x > 0.0)
            this((int)x);
        else
            super();
    }

    SuperInitBad01(int x) {
        if (x > 0)
            super();
        else
            this((double)x);
    }
}
