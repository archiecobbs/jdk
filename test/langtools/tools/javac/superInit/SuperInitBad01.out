SuperInitBad01.java:21:13: compiler.err.recursive.ctor.invocation
1 error
